src/Build/Microsoft.Build.csproj

```diff
@@ -26,6 +26,10 @@
 
     <!-- Do not generate a warning that our 'stable' package should not have a prerelease dependency. -->
     <NoWarn>$(NoWarn);NU5104</NoWarn>
+
+    <!-- Fnv-1a 64-bit hash we use in BuildEventArgsWriter to deduplicate strings uses UInt64 which is not CLS-compliant -->
+    <NoWarn>$(NoWarn);CS3001;CS3002;CS3003</NoWarn>
```

Comment:
Wouldn't it be better to mark the assembly as not CLS compliant? I'm probably missing something here.

Reply:
No, I just don't know about CLS compliance? I'd be happy to. How do I do that?

Reply:
pragmas are messy and I'd have to do it in a few places. Any concerns with changing the assembly to CLSCompliant(false)?
https://github.com/dotnet/msbuild/blob/74eb87c20d2495f81788e1bc7bb27203474818f4/src/Build/AssemblyInfo.cs#L39

Unless I'm missing anything CLS compliance was never useful for anything.

Reply:
Actually turns out we don't need any of these.